src/models/AbstractWidget.php

```diff
@@ -0,0 +1,126 @@
+<?php namespace Leean\Widgets\Models;
+
+use Leean\Acf\All;
+
+/**
+ * Class AbstractWidget.
+ *
+ * @package Leean\Widgets\Models
+ */
+class AbstractWidget extends \WP_Widget
+{
+	/**
+	 * The slug.
+	 *
+	 * @var bool|string
+	 */
+	private $_slug = '';
+
+	/**
+	 * AbstractWidget constructor.
+	 *
+	 * @param string $title       Widget title
+	 * @param string $description Widget description.
+	 * @param bool   $slug		  Widget slug (use class name if not supplied).
+	 */
+	public function __construct( $title, $description, $slug = false ) {
```

Comment:
That's the definition of an abstract class if you are not going to use it as such, you should removed the prefix abstract. 

In several languages (if not in all) an Abstract class can't be instantiated. 

Some defs: 

> In programming languages, an abstract type is a type in a nominative type system that cannot be instantiated directly.

>  Classes defined as abstract may not be instantiated, and any class that contains at least one abstract method must also be abstract. 

> Abstract classes cannot be instantiated, but they can be subclassed.